### PR TITLE
Added Sensor Python interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

This PR is part of this mete ticket https://github.com/ignitionrobotics/sdformat/issues/931

Require
 - https://github.com/ignitionrobotics/sdformat/pull/982
 - https://github.com/ignitionrobotics/sdformat/pull/983
 - https://github.com/ignitionrobotics/sdformat/pull/971
 - https://github.com/ignitionrobotics/sdformat/pull/970
 - https://github.com/ignitionrobotics/sdformat/pull/968
 - https://github.com/ignitionrobotics/sdformat/pull/981
 - https://github.com/ignitionrobotics/sdformat/pull/973
 - https://github.com/ignitionrobotics/sdformat/pull/972
 - https://github.com/ignitionrobotics/sdformat/pull/984
 - https://github.com/ignitionrobotics/sdformat/pull/969
 - https://github.com/ignitionrobotics/sdformat/pull/951

## Summary

Added Sensor Python interface

## Test it

```python
from sdformat import Sensor

sensor = Sensor()
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: python/src/sdf/pySensor.cc [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+         "is not an Altimeter.")
+    .def("set_altimeter_sensor", &sdf::Sensor::SetAltimeterSensor,
+         "Set the altimeter sensor.")
+    .def("air_pressure_sensor",
+         pybind11::overload_cast<>(
+           &sdf::Sensor::AirPressureSensor),
+         pybind11::return_value_policy::reference_internal,
+         "Get the update rate in Hz.")
+    .def("set_air_pressure_sensor", &sdf::Sensor::SetAirPressureSensor,
+         "Set the altimeter sensor.")
```
Comment (azeey) [inline python/src/sdf/pySensor.cc:116]: Update documentation

File: python/src/sdf/pySensor.cc [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+    .def("camera_sensor",
+         pybind11::overload_cast<>(
+           &sdf::Sensor::CameraSensor),
+         pybind11::return_value_policy::reference_internal,
+         "Get the update rate in Hz.")
```
Comment (azeey) [inline python/src/sdf/pySensor.cc:121]: Update documentation

File: python/src/sdf/pySensor.cc [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    .def("set_camera_sensor", &sdf::Sensor::SetCameraSensor,
+         "Set the altimeter sensor.")
```
Comment (azeey) [inline python/src/sdf/pySensor.cc:123]: Update documentation for this and the following functions.

File: python/test/pySensor_TEST.py [hunk 4]
```diff
@@ -0,0 +1,344 @@
...
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+from ignition.math import Pose3d
+from sdformat import AirPressure, Altimeter, Camera, IMU, ForceTorque, Lidar, Magnetometer, NavSat, Noise, SemanticPose, Sensor
```
Comment (azeey) [inline python/test/pySensor_TEST.py:17]: nit: Line too long.

File: python/test/pySensor_TEST.py [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+import unittest
+
+
+class SensorTEST(unittest.TestCase):
+
+  def test_default_construction(self):
```
Comment (azeey) [inline python/test/pySensor_TEST.py:23]: Indent 4 lines.

**Review by azeey — APPROVED** [review]
Formatting and documentation issues. Otherwise, LGTM!

File: python/test/pySensor_TEST.py (hunk 5 above)
Comment (ahcorde) [reply python/test/pySensor_TEST.py]: https://github.com/gazebosim/sdformat/pull/988/commits/9c31ba0aadb4ed5f183bf46b05c639bcb3008939

File: python/test/pySensor_TEST.py (hunk 4 above)
Comment (ahcorde) [reply python/test/pySensor_TEST.py]: https://github.com/gazebosim/sdformat/pull/988/commits/9c31ba0aadb4ed5f183bf46b05c639bcb3008939

File: python/src/sdf/pySensor.cc (hunk 2 above)
Comment (ahcorde) [reply python/src/sdf/pySensor.cc]: https://github.com/gazebosim/sdformat/pull/988/commits/9c31ba0aadb4ed5f183bf46b05c639bcb3008939

File: python/src/sdf/pySensor.cc (hunk 1 above)
Comment (ahcorde) [reply python/src/sdf/pySensor.cc]: https://github.com/gazebosim/sdformat/pull/988/commits/9c31ba0aadb4ed5f183bf46b05c639bcb3008939

**Pull request merged**